inference-engine/src/gna_plugin/transformations/broadcast_const.cpp

```diff
@@ -0,0 +1,171 @@
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include "legacy/ngraph_ops/eltwise.hpp"
+#include "ngraph_ops/convolution_ie.hpp"
+#include "ngraph/runtime/reference/broadcast.hpp"
+
+#include <vector>
+#include <functional>
+#include <numeric>
+
+using namespace GNAPluginNS;
+
+NGRAPH_RTTI_DEFINITION(BroadcastConst, "BroadcastConst", 0);
+
+using Node = std::shared_ptr<ngraph::Node>;
+using Nodes = std::vector<Node>;
+using Input = ngraph::Input<ngraph::Node>;
+using Inputs = std::vector<Input>;
+using AxisSet = std::set<size_t>;
+
+namespace {
+
+std::string Shape2Str(const ngraph::Shape & shape) {
+    std::string result;
+    for (auto value : shape) {
+        result += std::to_string(value);
+        result += " ";
+    }
+    return result;
+}
+
+Node CreateTiledConst(Node const_node, const ngraph::Shape & shape) {
+    std::shared_ptr<ngraph::opset8::Constant> old_const_node = std::dynamic_pointer_cast<ngraph::opset8::Constant>(const_node);
+
+    const ngraph::Shape const_node_shape = const_node->get_output_tensor(0).get_shape();
+    const auto elem_type = old_const_node->get_element_type();
+    std::vector<char> new_const_values(elem_type.size() * ngraph::shape_size(shape), 0);
+
+    AxisSet broadcast_axes;
+    ngraph::Shape new_const_shape = const_node_shape;
+    new_const_shape.insert(new_const_shape.begin(), shape.size() - new_const_shape.size(), 1);
+    for (int i = 0; i < new_const_shape.size(); ++i) {
+        if (new_const_shape[i] == shape[i])
+            continue;
+        if (shape[i] % new_const_shape[i]) {
+            std::cerr << "incompatible shapes new shape {" << Shape2Str(new_const_shape) << "} target shape {" << Shape2Str(shape) << "}" << std::endl;
+            return {};
+        }
+        broadcast_axes.insert(i);
+    }
+
+    ngraph::runtime::reference::broadcast(static_cast<const char*>(old_const_node->get_data_ptr()),
+                        new_const_values.data(),
+                        new_const_shape,
+                        shape,
+                        broadcast_axes,
+                        elem_type.size());
+
+    return ngraph::opset8::Constant::create(old_const_node->get_output_element_type(0),
+                                            shape,
+                                            new_const_values.data());
+}
+
+#ifdef GNA_LEGACY
+bool HasConvolutionInput(Node eltwise_node) {
+    const auto left_parent_node = eltwise_node->input_value(0).get_node_shared_ptr();
+    const auto right_parent_node = eltwise_node->input_value(1).get_node_shared_ptr();
+
+    return (IsAnyOfLayerTypes<ngraph::op::ConvolutionIE,
+                               ngraph::opset8::Convolution>(left_parent_node) ||
+            IsAnyOfLayerTypes<ngraph::op::ConvolutionIE,
+                               ngraph::opset8::Convolution>(right_parent_node));
+}
+
+bool IsFusedBiasConst(Node const_node) {
```

Comment:
I'd also check if FullyConnectedBiasFusion isn't broken, just to be sure

Reply:
it's not needed here any longer